app/controllers/concerns/collection.rb

```diff
@@ -29,9 +29,15 @@ def search(data)
     end
 
     def sort(data)
-      return data if params[:sort_by].blank?
-
-      data.order(build_order_by(data.klass, params[:sort_by]))
+      order = if params[:sort_by].present?
+                build_order_by(data.klass, params[:sort_by])
+              else
+                {}
+              end
+
+      # add id for deterministic pagination
+      order[:id] = :asc unless order.include?(:id)
```

Comment:
I guess we need the same for GraphQL.

Reply:
You could move the whole thing into the `build_order_by`, this way it would also propagate to GQL

Reply:
Also I think it logically belongs there

Reply:
Yes, I had that thought from the beginning. :slightly_smiling_face: